common/statistics/src/clients/mod.rs

```diff
@@ -0,0 +1,149 @@
+pub type ClientStatsReceiver = tokio::sync::mpsc::UnboundedReceiver<ClientStatsEvents>;
+
+/// Channel allowing generic statistics events to be reported to a stats event aggregator
+#[derive(Clone)]
+pub struct ClientStatsSender {
+    stats_tx: UnboundedSender<ClientStatsEvents>,
+}
+
+impl ClientStatsSender {
+    /// Create a new statistics Sender
+    pub fn new(stats_tx: UnboundedSender<ClientStatsEvents>) -> Self {
+        ClientStatsSender { stats_tx }
+    }
+
+    /// Report a statistics event using the sender.
+    pub fn report(&self, event: ClientStatsEvents) {
+        if let Err(err) = self.stats_tx.send(event) {
+            log::error!("Failed to send stats event: {:?}", err);
+        }
+    }
+
+    /// Used when stats reporting is disabled -- reads all incoming messages and discards them
+    pub fn sink(mut shutdown: nym_task::TaskClient) -> Self {
+        let (stats_tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+
+        //let's not propagate a shutdown if we happen to error out while doing the blackhole
+        shutdown.disarm();
+
+        spawn(async move {
+            loop {
+                tokio::select! {
+                    m = rx.recv() => {
+                        if m.is_none() {
+                            log::trace!("StatisticsSink: channel closed shutting down");
+                            break;
+                        }
+                    },
+                    _ = shutdown.recv_with_delay() => {
+                        log::trace!("StatisticsSink: Received shutdown");
+                        break;
+                    },
+                }
+            }
+            log::debug!("StatsSink: Exited");
+        });
+        Self { stats_tx }
+    }
+}
+
+/// Client Statistics events (static for now)
+pub enum ClientStatsEvents {
+    /// Packet count events
+    PacketStatistics(packet_statistics::PacketStatisticsEvent),
+    /// Gateway Connection events
+    GatewayConn(gateway_conn_statistics::GatewayStatsEvent),
+    /// Nym API connection events
+    NymApi(nym_api_statistics::NymApiStatsEvent),
+}
+
+/// Controls stats event handling and reporting
+pub struct ClientStatsController {
+    //static infos
+    last_update_time: OffsetDateTime,
+    client_id: String,
+    client_type: String,
+    os_information: os_info::Info,
+
+    // stats collection modules
+    packet_stats: packet_statistics::PacketStatisticsControl,
+    gateway_conn_stats: gateway_conn_statistics::GatewayStatsControl,
+    nym_api_stats: nym_api_statistics::NymApiStatsControl,
+}
+
+impl ClientStatsController {
+    /// Creates a ClientStatsController given a client_id
+    pub fn new(client_id: String, client_type: String) -> Self {
+        ClientStatsController {
+            //Safety : 0 is always a valid number of seconds
+            #[allow(clippy::unwrap_used)]
+            last_update_time: OffsetDateTime::now_utc().replace_second(0).unwrap(), // allow a bigger anonymity set wrt
```

Comment:
will this really make the anonimity set bigger given it will have a pretty unique values for micro and nanoseconds?

Reply:
The idea is more to hide the exact moment the message was sent. Delays in the mixnet pretty much make the micro and nano second info useless, but the second info can still be leveraged by a GPA I reckon, hence this